From bbb2db6adad14ffe626661b0e2781638ef0349d2 Mon Sep 17 00:00:00 2001
From: "neeraj.dantu" <dantuguf14105@gmail.com>
Date: Sun, 5 Dec 2021 16:16:01 -0600
Subject: [PATCH] sdcard-script

---
 create_sdcard_from_flashlayout.sh | 25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/create_sdcard_from_flashlayout.sh b/create_sdcard_from_flashlayout.sh
index 8095de6..ba8b4d3 100755
--- a/create_sdcard_from_flashlayout.sh
+++ b/create_sdcard_from_flashlayout.sh
@@ -35,10 +35,10 @@ SDCARD_TOKEN=mmc0
 # Size of 2GB
 #DEFAULT_RAW_SIZE=2048
 # Size of 1.5GB
-DEFAULT_RAW_SIZE=1536
+#DEFAULT_RAW_SIZE=1536
 
 # size of 768MB
-DEFAULT_ROOTFS_PARTITION_SIZE=763904
+#DEFAULT_ROOTFS_PARTITION_SIZE=763904
 # size of 1024MB
 #DEFAULT_ROOTFS_PARTITION_SIZE=1232896
 
@@ -328,15 +328,15 @@ function generate_gpt_partition_table_from_flash_layout() {
 			next_offset=${FLASHLAYOUT_data[$((i+1)),$COL_OFFSET]}
 			next_offset=${next_offset//0x/}
 			next_offset_b=$((16#$next_offset))
-			if [ "$partName" == "rootfs" ];
-			then
-				#force the size of rootfs parition to 768MB
-				new_next_partition_offset_b=$((offset_b + 1024*DEFAULT_ROOTFS_PARTITION_SIZE))
-				next_offset_b=$new_next_partition_offset_b
+#			if [ "$partName" == "rootfs" ];
+#			then
+#				#force the size of rootfs parition to 768MB
+#				new_next_partition_offset_b=$((offset_b + 1024*DEFAULT_ROOTFS_PARTITION_SIZE))
+#				next_offset_b=$new_next_partition_offset_b
 
-				move_partition_offset $((i+1)) $new_next_partition_offset_b
-				index_of_rootfs=$i
-			fi
+#				move_partition_offset $((i+1)) $new_next_partition_offset_b
+#				index_of_rootfs=$i
+#			fi
 
 			if [ $i -gt $index_of_rootfs ];
 			then
@@ -792,7 +792,7 @@ function print_warning() {
 function usage() {
 	echo ""
 	echo "Help:"
-	echo "   $0 <FlashLayout file>"
+	echo "   $0 <FlashLayout file> <sd card size>"
 	echo ""
 	exit 1
 }
@@ -800,7 +800,7 @@ function usage() {
 #        Main
 # ------------------
 # check opt args
-if [ $# -ne 1 ];
+if [ $# -ne 2 ];
 then
 	echo "[ERROR]: bad number of parameters"
 	echo ""
@@ -808,6 +808,7 @@ then
 else
 	tools_check
 
+	DEFAULT_RAW_SIZE=$2
 	FLASHLAYOUT_filename=$1
 	FLASHLAYOUT_filename_path=$(dirname "$FLASHLAYOUT_filename")
 	FLASHLAYOUT_filename_name=$(basename "$FLASHLAYOUT_filename")
-- 
2.17.1

